src/components/modals/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { ButtonWithModal } from './ButtonWithModal';
+import { ActionButton } from './ModalWithActionButtons';
+
+export interface ButtonConfirmationModalProps {
+  modalTitle: string;
+  modalContentText: string;
+  name: string;
+  confirmButtonProps: ActionButton;
+  cancelButtonProps: ActionButton;
+}
+
+export const ButtonWithConfirmationModal = ({
+  modalTitle,
+  modalContentText,
+  name,
+  confirmButtonProps,
+  cancelButtonProps,
+  ...otherOpenButtonProps
+}: ButtonConfirmationModalProps) => {
+  return (
+    <ButtonWithModal
+      modalTitle={modalTitle}
+      modalContentText={modalContentText}
+      name={name}
```

Comment:
i think we should do text as children for consistency. see what you wrote in ModalWithActionButtons for example.

essentially `<Button>buttontext</Button>`

Reply:
For ModalWithActionButtons, I'm still passing name from an object of props down to Button, so I think this is fine

Reply:
alright that makes sense - I think I do name through props for my custom buttons as well; we'll just have to standardize it later as ik in a couple places where we directly use the mui button we do the text as children

Reply:
Sounds good